### PR TITLE
Implement ability to use styles for login & logout page editor

### PR DESCRIPTION
**Pull request by mBeym**
This PR introduces content styles for the login/logout-page `IPE` content.

See:
- https://docu.ilias.de/go/wiki/wpage_8269_1357
- https://docu.ilias.de/go/wiki/wpage_7330_1357

### CODE REVIEW

File: components/ILIAS/Init/classes/class.ilStartUpGUI.php [hunk 1]
```diff
@@ -951,6 +951,11 @@ private function getLoginPageEditorHTML(): string
             return '';
         }
 
+        $this->dic->contentStyle()->gui()->addCss($this->mainTemplate, $this->getAuthSettingsRefId());
+        //$this->mainTemplate->setCurrentBlock('SyntaxStyle');
+        //$this->mainTemplate->setVariable('LOCATION_SYNTAX_STYLESHEET', ilObjStyleSheet::getSyntaxStylePath());
+        //$this->mainTemplate->parseCurrentBlock();
+
         // get page object
```
Comment (mjansenDatabay) [inline components/ILIAS/Init/classes/class.ilStartUpGUI.php]:
```suggestion

```

File: components/ILIAS/Authentication/classes/class.ilObjAuthSettingsGUI.php [hunk 2]
```diff
@@ -859,7 +865,19 @@ public function executeCommand(): void
...
+                $this->setSubTabs("authSettings");
+                $this->tabs_gui->activateTab('authentication_settings');
+                $this->tabs_gui->activateSubTab("style");
+                $settings_gui = $this->content_style_gui
+                    ->objectSettingsGUIForRefId(
+                        null,
+                        $this->object->getRefId()
+                    );
+                $this->ctrl->forwardCommand($settings_gui);
+                break;
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/class.ilObjAuthSettingsGUI.php:881]:
```suggestion
                break;

```

File: components/ILIAS/Authentication/classes/class.ilObjAuthSettingsGUI.php [hunk 3]
```diff
@@ -986,6 +1004,14 @@ public function setSubTabs(string $a_tab): void
                     $this->ctrl->getLinkTargetByClass(ilAuthLogoutPageEditorGUI::class)
                 );
             }
+
+            if ($this->access->checkAccess('write', '', $this->object->getRefId())) {
+                $this->tabs_gui->addSubTab(
+                    self::UI_TAB_ID_STYLE,
+                    $this->lng->txt('cont_style'),
+                    $this->ctrl->getLinkTargetByClass(ilObjectContentStyleSettingsGUI::class, "")
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/class.ilObjAuthSettingsGUI.php]:
```suggestion
                    $this->ctrl->getLinkTargetByClass(ilObjectContentStyleSettingsGUI::class)
```

File: components/ILIAS/Authentication/classes/class.ilObjAuthSettingsGUI.php [hunk 4]
```diff
@@ -859,7 +865,19 @@ public function executeCommand(): void
                 $lpe = new ilAuthLogoutPageEditorGUI($this->object->getRefId());
                 $this->ctrl->forwardCommand($lpe);
                 break;
-
+            case strtolower(ilObjectContentStyleSettingsGUI::class):
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/class.ilObjAuthSettingsGUI.php:869]:
```suggestion

            case strtolower(ilObjectContentStyleSettingsGUI::class):
```

File: components/ILIAS/Authentication/classes/class.ilObjAuthSettingsGUI.php [hunk 5]
```diff
@@ -859,7 +865,19 @@ public function executeCommand(): void
...
+                $this->checkPermission("write");
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/class.ilObjAuthSettingsGUI.php]:
```suggestion
                $this->checkPermission('write');
```

File: components/ILIAS/Authentication/classes/class.ilObjAuthSettingsGUI.php [hunk 6]
```diff
@@ -859,7 +865,19 @@ public function executeCommand(): void
...
+                $this->setTitleAndDescription();
+                $this->setSubTabs("authSettings");
+                $this->tabs_gui->activateTab('authentication_settings');
+                $this->tabs_gui->activateSubTab("style");
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/class.ilObjAuthSettingsGUI.php]:
```suggestion
                $this->setSubTabs('authSettings');
                $this->tabs_gui->activateTab('authentication_settings');
                $this->tabs_gui->activateSubTab('style');
```

File: components/ILIAS/Authentication/classes/class.ilAuthLoginPageEditorGUI.php [hunk 7]
```diff
@@ -53,9 +57,13 @@ public function __construct(int $a_ref_id)
...
 
-        $this->content_style_domain = $DIC->contentStyle()
-                                          ->domain()
-                                          ->styleForRefId($a_ref_id);
+        $this->ref_id = $a_ref_id;
+
+        $content_style = $DIC->contentStyle();
+        $this->content_style_domain = $content_style
+            ->domain()
+            ->styleForRefId($a_ref_id);
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/class.ilAuthLoginPageEditorGUI.php:65]: Why do we use a different "ref_id" compared to `ilStartupGUI`.

File: components/ILIAS/Authentication/classes/class.ilAuthLogoutPageEditorGUI.php [hunk 8]
```diff
@@ -65,9 +68,12 @@ public function __construct(int $a_ref_id)
...
 
         $this->settings = ilAuthLogoutPageEditorSettings::getInstance();
-        $this->content_style_domain = $DIC->contentStyle()
-                                          ->domain()
-                                          ->styleForRefId($a_ref_id);
+
+        $content_style = $DIC->contentStyle();
+        $this->content_style_domain = $content_style
+            ->domain()
+            ->styleForRefId($a_ref_id);
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/class.ilAuthLogoutPageEditorGUI.php:75]: Why do we use a different "ref_id" compared to `ilStartupGUI`.

File: components/ILIAS/Authentication/classes/class.ilAuthLogoutPageEditorGUI.php [hunk 9]
```diff
@@ -143,6 +149,7 @@ private function forwardToPageObject(): void
 
         $this->tpl->addCss(ilObjStyleSheet::getContentStylePath(0));
         $this->tpl->addCss(ilObjStyleSheet::getSyntaxStylePath());
+        $this->content_style_gui->addCss($this->tpl,  $this->ref_id);
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/class.ilAuthLogoutPageEditorGUI.php]: Why do we use a different "ref_id" compared to `ilStartupGUI`.

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

File: components/ILIAS/Authentication/classes/class.ilAuthLoginPageEditorGUI.php (hunk 7 above)
Comment (mBeym) [reply components/ILIAS/Authentication/classes/class.ilAuthLoginPageEditorGUI.php:65]: why should I load the ref_id from the database when it's already supplied as a parameter?

Comment (mjansenDatabay) [reply components/ILIAS/Authentication/classes/class.ilAuthLoginPageEditorGUI.php:65]: If it is the identical `$ref_id` we could leave this as it is.

File: components/ILIAS/Init/classes/class.ilStartUpGUI.php [hunk 10]
```diff
@@ -987,6 +989,19 @@ private function getLogoutPageEditorHTML(): string
...
+        $auth_settings_obj_data = $this->dic->database()->fetchAssoc(
+            $this->dic->database()->queryF(
+                "SELECT ref_id FROM object_reference WHERE obj_id = (SELECT obj_id FROM object_data WHERE type = %s)",
+                [ilDBConstants::T_TEXT],
+                ['auth']
+            )
+        );
+
+        return (int) $auth_settings_obj_data['ref_id'];
+    }
```
Comment (mjansenDatabay) [inline components/ILIAS/Init/classes/class.ilStartUpGUI.php]: This method must not be part of a GUI class.

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

File: components/ILIAS/Init/classes/class.ilStartUpGUI.php [hunk 11]
```diff
@@ -951,7 +951,14 @@ private function getLoginPageEditorHTML(): string
...
         }
 
-        // get page object
+        $auth_settings_ref_id = (int) $this->dic->database()->fetchAssoc(
+            $this->dic->database()->queryF(
+                "SELECT ref_id FROM object_reference WHERE obj_id = (SELECT obj_id FROM object_data WHERE type = %s)",
+                [ilDBConstants::T_TEXT],
+                ['auth']
+            )
+        )['ref_id'];
```
Comment (mjansenDatabay) [inline components/ILIAS/Init/classes/class.ilStartUpGUI.php]: We MUST NOT put any database operations in the GUI-layer.

File: components/ILIAS/Init/classes/class.ilStartUpGUI.php [hunk 12]
```diff
@@ -976,6 +983,16 @@ private function getLogoutPageEditorHTML(): string
             return '';
         }
 
+        $auth_settings_ref_id = (int) $this->dic->database()->fetchAssoc(
+            $this->dic->database()->queryF(
+                "SELECT ref_id FROM object_reference WHERE obj_id = (SELECT obj_id FROM object_data WHERE type = %s)",
+                [ilDBConstants::T_TEXT],
+                ['auth']
+            )
+        )['ref_id'];
```
Comment (mjansenDatabay) [inline components/ILIAS/Init/classes/class.ilStartUpGUI.php]: We MUST NOT put any database operations in the GUI-layer.

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

File: components/ILIAS/Init/classes/class.ilStartUpGUI.php [hunk 13]
```diff
@@ -983,14 +981,10 @@ private function getLogoutPageEditorHTML(): string
...
-                "SELECT ref_id FROM object_reference WHERE obj_id = (SELECT obj_id FROM object_data WHERE type = %s)",
-                [ilDBConstants::T_TEXT],
-                ['auth']
-            )
-        )['ref_id'];
-
+        $auth_settings_objects = ilObject::_getObjectsByType('auth');
+        $auth_settings_obj_id = (int) reset($auth_settings_objects)['obj_id'];
+        $auth_settings_ref_ids = ilObject::_getAllReferences($auth_settings_obj_id);
+        $auth_settings_ref_id = (int) reset($auth_settings_ref_ids);
```
Comment (mjansenDatabay) [inline components/ILIAS/Init/classes/class.ilStartUpGUI.php]: This looks like duplicate code, please provide a centralized method for this.

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

File: components/ILIAS/Authentication/classes/class.ilObjAuthSettings.php [hunk 14]
```diff
@@ -72,4 +72,12 @@ public function checkAuthScript(): bool
 
         return true;
     }
+
+    public static function getAuthSettingsRefId(): int
+    {
+        $auth_settings_objects = ilObject::_getObjectsByType('auth');
+        $auth_settings_obj_id = (int) reset($auth_settings_objects)['obj_id'];
+        $auth_settings_ref_ids = ilObject::_getAllReferences($auth_settings_obj_id);
+        return (int) reset($auth_settings_ref_ids);
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/class.ilObjAuthSettings.php:82]:
```suggestion

        return (int) reset($auth_settings_ref_ids);
```

File: components/ILIAS/Authentication/classes/class.ilObjAuthSettingsGUI.php [hunk 15]
```diff
@@ -18,21 +18,25 @@
...
 /**
  * @author Sascha Hofmann <saschahofmann@gmx.de>
  *
  * @ilCtrl_Calls ilObjAuthSettingsGUI: ilPermissionGUI, ilRegistrationSettingsGUI, ilLDAPSettingsGUI
  * @ilCtrl_Calls ilObjAuthSettingsGUI: ilAuthShibbolethSettingsGUI, ilCASSettingsGUI
  * @ilCtrl_Calls ilObjAuthSettingsGUI: ilSamlSettingsGUI, ilOpenIdConnectSettingsGUI
+ * @ilCtrl_Calls ilObjAuthSettingsGUI: ilObjectContentStyleSettingsGUI
  */
-class ilObjAuthSettingsGUI extends ilObjectGUI
+class ilObjAuthSettingsGUI extends ilObjectGUI implements ilContentPageObjectConstants
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/class.ilObjAuthSettingsGUI.php]:
```suggestion
class ilObjAuthSettingsGUI extends ilObjectGUI
```

File: components/ILIAS/Authentication/classes/class.ilObjAuthSettingsGUI.php [hunk 16]
```diff
@@ -986,6 +1006,14 @@ public function setSubTabs(string $a_tab): void
                     $this->ctrl->getLinkTargetByClass(ilAuthLogoutPageEditorGUI::class)
                 );
             }
+
+            if ($this->access->checkAccess('write', '', $this->object->getRefId())) {
+                $this->tabs_gui->addSubTab(
+                    self::UI_TAB_ID_STYLE,
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/class.ilObjAuthSettingsGUI.php]:
```suggestion
                    'style',
```

**Pull request merged**